src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepositoryImpl.java

```diff
@@ -416,11 +379,246 @@ public PassingRateStatisticsResult passingRateStatistics(Filter filter, Sort sor
 				.fetchInto(FlakyCasesTableContent.class);
 	}
 
-	private static Field<?> fieldName(TableField tableField) {
-		return field(name(tableField.getName()));
+	@Override
+	public Map<String, List<LaunchesStatisticsContent>> cumulativeTrendStatistics(Filter filter, List<String> contentFields, Sort sort,
+			String tagPrefix, int limit) {
+
+		List<Field<?>> fields = buildFieldsFromContentFields(contentFields);
+
+		Collections.addAll(fields,
+				fieldName(LAUNCHES_SUB_QUERY, ID),
+				fieldName(LAUNCH.NUMBER),
+				fieldName(LAUNCH.START_TIME),
+				fieldName(LAUNCH.NAME),
+				fieldName(LAUNCH_TAG.VALUE).as(TAG_VALUE)
+		);
+
+		Select<Record> select = dsl.select(fields)
+				.from(QueryBuilder.newBuilder(filter).with(sort).with(LAUNCHES_COUNT).build().asTable(LAUNCHES_SUB_QUERY))
+				.join(LAUNCH_TAG)
+				.on(LAUNCH_TAG.LAUNCH_ID.eq(fieldName(LAUNCHES_SUB_QUERY, ID).cast(Long.class)))
+				.where(LAUNCH_TAG.VALUE.in(DSL.selectDistinct(LAUNCH_TAG.VALUE)
+						.on(LAUNCH_TAG.VALUE)
+						.from(LAUNCH_TAG)
+						.where(LAUNCH_TAG.VALUE.like(tagPrefix + LIKE_CONDITION_SYMBOL))
+						.orderBy(LAUNCH_TAG.VALUE.desc())
+						.limit(limit)))
+				.orderBy(ofNullable(sort).map(s -> StreamSupport.stream(s.spliterator(), false)
+						.map(order -> field(name(order.getProperty())).sort(order.getDirection().isDescending() ?
```

Comment:
check this for trending widgets. they should show only last results after applying limit